contracts/test/SequencerInbox.t.sol

```diff
@@ -82,4 +126,85 @@ contract SequencerInboxTest is BaseSetup {
         uint256[] memory txLengths = new uint256[](1);
         seqIn.appendTxBatch(contexts, txLengths, "0x");
     }
+
+    /////////////////////////////
+    // appendTxBatch
+    /////////////////////////////
+    function test_appendTxBatch_positiveCase() public {
+        uint256 inboxSizeInitial = seqIn.getInboxSize();
+
+        // Each context corresponds to a single "L2 block"
+        // `contexts` is represented with uint256 3-tuple: (numTxs, l2BlockNumber, l2Timestamp)
+        // Let's create an array of contexts
+        uint256 numTxns = 3;
+        uint256 timeStamp1 = block.timestamp / 10;
+        uint256 timeStamp2 = block.timestamp / 5;
+        uint256 blockNumber1 = timeStamp1 / 20;
+        uint256 blockNumber2 = timeStamp2 / 20;
+
+        uint256[] memory contexts = new uint256[](6);
+        uint256[] memory txLengths = new uint256[](6);
+
+        // Let's assume that we had 2 blocks and each had 3 transactions
+        contexts[0] = (numTxns);
+        contexts[1] = (blockNumber1);
+        contexts[2] = (timeStamp1);
+        contexts[3] = (numTxns);
+        contexts[4] = (blockNumber2);
+        contexts[5] = (timeStamp2);
+
+        // txLengths is defined as: Array of lengths of each encoded tx in txBatch
+        for (uint256 i; i < 6; i++) {
+            txLengths[i] = (uint256(268));
+        }
+
+        // txBatch is defined as: Batch of RLP-encoded transactions
+        bytes memory txBatch = createTxBatch();
+
+        // Pranking as the sequencer and calling appendTxBatch
+        vm.prank(sequencer);
+        seqIn.appendTxBatch(contexts, txLengths, txBatch);
+
+        uint256 inboxSizeFinal = seqIn.getInboxSize();
+
+        assertGt(inboxSizeFinal, inboxSizeInitial);
+        assertEq(inboxSizeFinal, uint256(6), "2 blocks of 3 txns each = 6 numTxns");
+    }
+
+    function test_appendTxBatch_incompleteDataInContextsArray() public {
+        uint256 inboxSizeInitial = seqIn.getInboxSize();
+
+        // Each context corresponds to a single "L2 block"
+        // `contexts` is represented with uint256 3-tuple: (numTxs, l2BlockNumber, l2Timestamp)
+        // Let's create an array of contexts
+        uint256 numTxns = 3;
+        uint256 timeStamp1 = block.timestamp / 10;
+        uint256 blockNumber1 = timeStamp1 / 20;
+
+        uint256[] memory contexts = new uint256[](4);
+        uint256[] memory txLengths = new uint256[](6);
```

Comment:
define all related constants together (e.g. this is 6 because of how you've written `createTxBatch` but if you add a tx there this doesn't change)

Reply:
Sure, will come up with a better structure for this.